aioxmpp/bookmarks/__init__.py

```diff
@@ -37,50 +38,36 @@
 XSOs
 ====
 
-The following XSOs are used to represent an manipulate bookmark lists.
+All bookmark types must adhere to the following ABC:
 
-.. autoclass:: Storage
+.. autoclass:: Bookmark
+
+The following XSOs are used to represent an manipulate bookmark lists.
 
 .. autoclass:: Conference
 
 .. autoclass:: URL
 
-Notes on usage
-==============
-
-It is important to use this class carefully to prevent race conditions
-with other clients modifying the bookmark data (unfortunately, this is
-not entirely preventable due to the design of the bookmark protocol).
+The following is used internally as the XSO container for bookmarks. If
+you want to support custom bookmark types, they have to be registered
+as children with the :attr:`Storage.bookmarks` descriptor:
 
-The recommended practice is to modify the bookmarks in a get, modify,
-set fashion, where the modify step should be as short as possible
-(that is, should not depend on user input).
-
-The individual bookmark classes support comparison by value. This
-allows useful manipulation of the bookmark list in agreement with
-the get-modify-set pattern. Where we reference the objects to operate
-on by with the value retrieved in an earlier get.
-
-Removal::
-
-  storage = bookmark_client.get_bookmarks()
-  storage.bookmarks.remove(old_bookmark)
-  bookmark_client.set_bookmarks(storage)
+.. autoclass:: Storage
 
-Modifying::
+Notes on usage
+==============
 
-  storage = bookmark_client.get_bookmarks()
-  i = storage.bookmarks.index(old_bookmark)
-  storage.bookmarks[i].name = "New Shiny Name"
-  storage.bookmarks[i].nick = "new_nick"
-  bookmark_client.set_bookmarks(storage)
+.. currentmodule:: aioxmpp
```

Comment:
BookmarkClient does not resolve via aioxmpp.bookmarks (I don't get why, but compare above, where `.. currentmodule:: aioxmpp` is used to document BookmarkClient).